napari/utils/_testsupport.py

```diff
@@ -228,7 +241,14 @@ def actual_factory(
     if request.config.getoption(_SAVE_GRAPH_OPNAME):
         fail_obj_graph(QtViewer)
 
+    if request.node.rep_call.failed:
+        # IF test failed do not check for leaks
+        QtViewer._instances.clear()
+
     _do_not_inline_below = len(QtViewer._instances)
+
+    QtViewer._instances.clear()  # clear to prevent fail of next test
```

Comment:
I'm not sure why i didn't do that originally.

Reply:
Me too